Routes/doctor.js

```diff
@@ -0,0 +1,437 @@
+ *      description: Add a prescription for a patient
+ *      requestBody:
+ *          required: true
+ *          content:
+ *              application/json:
+ *                  schema:
+ *                      $ref: '#/components/schemas/Prescription'
+ *      responses:
+ *          200:
+ *              description: Added Successfully
+ *              
+ */ 
+
+DoctorRoute.post('/addPrescription',verifyToken, async(req,res) => {
+    const patient = await Patients.findById(req.body.patientId)
+   
+    const prescription = new Prescription({
+        doctorId: res.current_user.Doctor._id,
+        patientId: req.body.patientId,
+        doctorName: res.current_user.Doctor.doctorName,
+        patientName: patient.patientName,
+        diagnose: req.body.diagnose,
+        // medicineId: req.body.medicineId,
+    })
+    // const addDoctor = await patient.findOneAndUpdate({_id:req.body.patientId},{doctorId: res.current_user.Doctor._id}, {new: true})
+    const newprescription = await prescription.save()
+    res.json(prescription); 
+    // res.json(res.current_user.Doctor._id)
+
+})
+
+/**
+ * @swagger
+ * /doctor/addPrescription/addMedicine:
+ *  post:
+ *      summary: Add medicine for a prescription of a  patient  
+ *      description: Add medicine for a prescription of a  patient
+ *      requestBody:
+ *          required: true
+ *          content:
+ *              application/json:
+ *                  schema:
+ *                      $ref: '#/components/schemas/PrescriptionMedicine'
+ *      responses:
+ *          200:
+ *              description: Added Successfully
+ *              
+ */ 
+
+
+DoctorRoute.post('/addPrescription/addMedicine',verifyToken,async(req,res) => {
+    const prescription = await Prescription.findById(req.body.prescriptionId)
+    
+    if (prescription != null || prescription != undefined){
+        
+        const medicine = await Medicine.findById(req.body.medicineId)
+        if (medicine != null || medicine != undefined){
+            prescription.medicines.push(medicine) 
+            let updatedPrescription = await prescription.save()
+            res.json(updatedPrescription)
+            
+        }
+        else{
+            res.json({result: "medicine not found"})
+        }
+    }
+    else{
+        res.json({result: "prescription not found"})
+    }
+})
+
+/** 
+* @swagger
+* /doctor/bill:
+*  post:
+*      summary: generate bill for a patient on basis of prescription id 
+*      description: generate bill for a patient on basis of prescription id 
+*      requestBody:
+*          required: true
+*          content:
+*              application/json:
+*                  schema:
+*                      $ref: '#/components/schemas/bill'
+*      responses:
+*          200:
+*              description: Added Successfully
+*              
+*/ 
+
+DoctorRoute.post('/bill', verifyToken, async (req,res) =>{
+    let prescription = await Prescription.findById(req.body.prescriptionId)
```

Comment:
Always make sure you have encapsulated your controller function within a `try{} catch(e){}` block so that any unwanted error doesn't lead to crash of server

Reply:
[Implemented](https://github.com/Sankalp-Sachapara/HospitalPractice/pull/2/files/893ce87fe4fe8a5be2c037597a3a6cfbdffc56ac#r1045073575)